File: test_fetch_size.py
import requests, time

res = requests.post("http://localhost:8001/api/v1/auth/login", data={"username": "testuser", "password": "testuser"})
token = res.json().get("access_token")

start = time.time()
res = requests.get("http://localhost:8001/api/v1/menus/items", headers={"Authorization": f"Bearer {token}"})
end = time.time()

print(f"Menus API size: {len(res.content)} bytes")
print(f"Time taken: {end - start:.3f} seconds")

start = time.time()
res = requests.get("http://localhost:8001/api/v1/orders/history", headers={"Authorization": f"Bearer {token}"})
end = time.time()

print(f"Orders API size: {len(res.content)} bytes")
print(f"Time taken: {end - start:.3f} seconds")

File: test_fetch_size.js
const http = require('http');

const data = JSON.stringify({
  taikhoan: 'KH-987E7770',
  matkhau: '123'
});

const req = http.request({
  hostname: '127.0.0.1',
  port: 8001,
  path: '/api/v1/auth/login',
  method: 'POST',
  headers: {
    'Content-Type': 'application/json',
    'Content-Length': data.length
  }
}, (res) => {
  let body = '';
  res.on('data', d => body += d);
  res.on('end', () => {
    const token = JSON.parse(body).accessToken;
    
    // Fetch menus
    http.get({
      hostname: '127.0.0.1',
      port: 8001,
      path: '/api/v1/menus/items',
      headers: { 'Authorization': `Bearer ${token}` }
    }, (res) => {
      let b = 0;
      res.on('data', d => b += d.length);
      res.on('end', () => console.log('Menus API size:', b));
    });
    
    // Fetch orders
    http.get({
      hostname: '127.0.0.1',
      port: 8001,
      path: '/api/v1/orders/history',
      headers: { 'Authorization': `Bearer ${token}` }
    }, (res) => {
      let b = 0;
      res.on('data', d => b += d.length);
      res.on('end', () => console.log('Orders API size:', b));
    });
  });
});
req.write(data);
req.end();
